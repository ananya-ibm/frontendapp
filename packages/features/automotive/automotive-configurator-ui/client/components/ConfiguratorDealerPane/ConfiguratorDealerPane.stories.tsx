/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { ArrowLeft, ArrowRight } from '@carbon/react/icons';
import { ConfiguratorDealerPane } from './ConfiguratorDealerPane';

export default {
  title: 'Features/Automotive/Configurator/Smart Components/Product/ConfiguratorDealerPane',
  component: ConfiguratorDealerPane,
  argTypes: {
    beforeIcon: { control: { disable: true } },
    afterIcon: { control: { disable: true } }
  }
};

const storyProps = {
  beforeIcon: <ArrowLeft size={32} />,
  text: 'Go back'
};

const storyProps2 = {
  afterIcon: <ArrowRight size={32} />,
  text: 'Next'
};

export const beforeIcon = args => <ConfiguratorDealerPane {...args} />;
beforeIcon.args = storyProps;

export const afterIcon = args => <ConfiguratorDealerPane {...args} />;
afterIcon.args = storyProps2;
