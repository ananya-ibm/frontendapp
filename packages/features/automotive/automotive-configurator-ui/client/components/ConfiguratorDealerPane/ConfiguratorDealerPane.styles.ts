/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import styled from 'styled-components';

export const ConfiguratorDealerPane = styled('div')`
  min-width: 30rem;
  padding-right: 2rem;
`;

export const Title = styled('h2')`
  padding-bottom: 1rem;
  padding-top: 2rem;
`;

export const ConfiguratorTitle = styled('h4')`
  padding-bottom: 0.7rem;
  padding-top: 2rem;
`;
