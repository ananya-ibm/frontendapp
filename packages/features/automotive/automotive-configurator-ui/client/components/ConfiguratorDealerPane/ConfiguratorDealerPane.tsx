/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import JSONTree from 'react-json-tree';
import { Button, TextInput } from '@exo/frontend-components-base';
import * as S from './ConfiguratorDealerPane.styles';

export const ConfiguratorDealerPane = ({
  type,
  handleCustomerIDChange,
  cusomterConfigurationId,
  categories,
  financeData,
  handleBackToCustomer
}: Props) => {
  return type === 'DEALER' ? (
    <S.ConfiguratorDealerPane style={{}}>
      <S.Title>Dealer view</S.Title>
      <TextInput
        labelText="Enter customer Configurator ID:"
        id="cusomterConfigurationIdInput"
        type="text"
        onChange={handleCustomerIDChange}
        value={cusomterConfigurationId ?? ''}
      />
      <S.ConfiguratorTitle>Configurator Status</S.ConfiguratorTitle>
      <JSONTree data={{ categories, financeData }} />
      <Button variant="secondary" label="Back to customer view" onClick={handleBackToCustomer} />
    </S.ConfiguratorDealerPane>
  ) : (
    <></>
  );
};

type Props = {
  type: string;
  handleCustomerIDChange: () => void;
  cusomterConfigurationId: string;
  categories: any;
  financeData: any;
  handleBackToCustomer: () => void;
};

export default ConfiguratorDealerPane;
