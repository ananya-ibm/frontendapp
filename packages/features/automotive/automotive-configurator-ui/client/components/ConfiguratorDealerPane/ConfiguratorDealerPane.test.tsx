/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { renderWithTheme as render } from '@testUtils';
import { ArrowLeft, ArrowRight } from '@carbon/react/icons';
import { ConfiguratorDealerPane } from './ConfiguratorDealerPane';

const testProps = {
  onClick: () => {},
  beforeIcon: <ArrowLeft size={32} />,
  afterIcon: <ArrowRight size={32} />,
  text: 'Next'
};

describe('<ConfiguratorDealerPane /> component', () => {
  test('its snapshot matches', () => {
    // @ts-ignore
    const { container } = render(<ConfiguratorDealerPane {...testProps} />);

    expect(container).toBeDefined();
    expect(container.firstChild).toMatchSnapshot();
  });
});
