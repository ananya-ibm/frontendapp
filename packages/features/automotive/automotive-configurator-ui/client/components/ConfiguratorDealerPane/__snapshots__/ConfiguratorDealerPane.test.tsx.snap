// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfiguratorDealerPane /> component its snapshot matches 1`] = `null`;
